templates/commands/render/action.go

```diff
@@ -141,6 +141,29 @@ func templateAndCompileRegexes(regexes []model.String, scope *common.Scope) ([]*
 	return compiled, merr
 }
 
+// processPaths processes a list of input String paths for go templating, relative paths,
+// OS-specific slashes, and (soon) file globbing.
+func processPaths(paths []model.String, scope *common.Scope) ([]model.String, error) {
+	pathsCopy := make([]model.String, len(paths))
+	copy(pathsCopy, paths)
+
+	var err error
+	for i, p := range pathsCopy {
+		p.Val, err = parseAndExecuteGoTmpl(p.Pos, p.Val, scope)
+		if err != nil {
+			return nil, err
+		}
+		p.Val = filepath.FromSlash(p.Val)
+		p.Val, err = safeRelPath(p.Pos, p.Val)
+		if err != nil {
+			return nil, err
+		}
+		pathsCopy[i].Val = p.Val
+	}
+
+	return pathsCopy, nil
+}
```

Comment:
Style nit: I think it's generally more common to create an empty slice and then append values while iterating over the given slice (instead of the current implementation which makes a copy of the slice and modifies it in place):

```go
	pathsCopy := make([]model.String, 0, len(paths))
	for _, p := range paths {
    val, err := parseAndExecuteGoTmpl(p.Pos, p.Val, scope)
		if err != nil {
			return nil, err
		}

    val, err = safeReplPath(p.Pos, filepath.FromSlash(val))
		if err != nil {
			return nil, err
		}

		pathsCopy = append(pathsCopy, model.String{
			Val: val,
			Pos: p.Pos,
			// ...
		})
	}

	return pathsCopy, nil
```

Reply:
Changed to append to empty slice.